Clarify SonarQube integration installation and configuration instructions (#8818)
DataDog/integrations-core: sonarqube/README.md

### Configuration

SonarQube exposes metrics from two sources: its web API and JMX. To collect all of the

> [metrics specified below](#metrics), you'll need to configure two intances of this check. One instance will monitor
> SonarQube's JMX beans and the other will monitor SonarQube's Web API.

Reviewer: ```suggestion
[metrics specified below](#metrics), configure two instances of this check, one to monitor
SonarQube's web API, and the other to monitor SonarQube's JMX beans.
```

> Documentation on SonarQube's Web API is available at `/web_api` on your SonarQube web UI. By default this integration

Reviewer: ```suggestion
Documentation on SonarQube's web API is available at `/web_api` on your SonarQube web UI. By default this integration
```

collects all relevant SonarQube performance metrics exposed through SonarQube's JMX beans. The configuration for these

> default metrics is available in the [sonarqube.d/metrics.yaml][12] file. Documentation on these beans is available on

Reviewer: should the link text here match the filename on github, which is `sonarqube/data/metrics.yaml`?

Author: When installed its right next to the `conf.yaml` -- the `data` directory doesn't exist

Author: ```
vagrant@sonarqube:~$ ll /etc/datadog-agent/conf.d/sonarqube.d/
-rw-r--r--   1 dd-agent dd-agent  1537 Mar 11 19:29 conf.yaml
-rw-r--r--   1 dd-agent dd-agent 21775 Feb 23 11:28 conf.yaml.example
-rw-r--r--   1 dd-agent dd-agent  2688 Feb 23 11:28 metrics.yaml
```

[SonarQube's website][13].

SonarQube's JMX server is not enabled by default. More information on how to enable and configure JMX within SonarQube
in available within the [SonarQube documentation][14].

> This is a basic `sonarqube.d/conf.yaml` example based on SonarQube and JMX defaults and can be used a starting point

Reviewer: ```suggestion
This is a basic `sonarqube.d/conf.yaml` example based on SonarQube and JMX defaults. You can use it as a starting point
```

when configuring for both the host-based or container-based agent installation.

```yaml
init_config:
    is_jmx: false
    collect_default_metrics: true
instances:
  # Web API instance
  - is_jmx: false
    web_endpoint: http://localhost:9000
    auth_type: basic
    username: <username> # Defined in the Web UI
    password: <password> # Defined in the Web UI
    components:
      my-project:
        tag: my-branch
  # JMX instance
  - host: localhost
    is_jmx: true
    port: 10443
    user: <username>      # Defined in SonarQube's sonar.properties file
    password: <password>  # Defined in SonarQube's sonar.properties file
```

> > Note: Once configured, you need to have Sonar scan at least one project for many of the metrics to populate in

Reviewer: ```suggestion
> Note: Once the integration is configured, have Sonar scan at least one project so that the metrics populate in
```

> Datadog.

> Projects in SonarQube often contain multiple SCM branches and tags. To specify what branch or tag for the integration

Reviewer: ```suggestion
Projects in SonarQube often contain multiple source control branches and tags. To specify which branch or tag for the integration
```

> to collect metrics from the Web UI, specify the `tag` configuration for the component, otherwise the default branch

Reviewer: ```suggestion
to collect metrics from the Web UI, specify the `tag` configuration for the component. Otherwise the default branch
```

> will be used (typically `master`).

Reviewer: ```suggestion
will be used (typically `main`).
```

Author: I know I'm assuming git here, but I've never heard of a `main` branch being used. Curious if this is specific to Sonar and I'm just lacking insight here

Reviewer: Github is enabling folks to rename their `master` branch to `main` (or whatever) in an effort to use more inclusive terminology. I'm just putting the wish out into the world that `master` gradually disappears, and `main` is normalized.

Author: Love it! I'll update it to read `main`.
